web/src/utils/dashboard/colorPalette.ts

```
@@ -0,0 +1,284 @@
+
+export const getMetricMinMaxValue = (searchQueryData: any) => {
+  // need min and max value for color
+  let min = Infinity;
+  let max = -Infinity;
+  searchQueryData.forEach((metric: any) => {
+    if (metric.result && Array.isArray(metric.result)) {
+      metric?.result?.forEach((valuesArr: any) => {
+        if (valuesArr.values && Array.isArray(valuesArr.values)) {
+          valuesArr.values.forEach((val: any) => {
+            // val[1] should not NaN
+            if (!isNaN(val[1])) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)